Permission denied error when shlink tries to download same GeoLite db more than once

<!--
Before opening an issue, just take into account that this is a completely free of charge and open source project.
I'm always happy to help and provide support, but some understanding will be expected.
I do this in my own free time, so expect some delays when implementing new features and fixing bugs, and don't take it personal if an issue gets eventually closed.
You may also be asked to provide tests or ways to reproduce reported bugs.
Try to be polite, and understand it is impossible for an OSS project to cover all use cases.

With that said, please fill in the information requested next. More information might be requested next (like logs or system configs).
-->

#### How Shlink is set-up

* Shlink Version: 2.4.0
* PHP Version: 7.4.12
* How do you serve Shlink: Self-hosted Apache
* Database engine used: MySQL

#### Summary

We have setup multiple installations on one server each served by a different vhost.
For geolocation we have configured cronjobs which run in the context of the user/group of the corresponding vhost.

#### Current behavior

When a Geolite2 database download is available and needs to be downloaded there is a conflict extracting the file Geolite2-City.mmdb file into the temporary directory. The first run of the task ./cli visit:locate creates a directory in the configured temp_dir "GeoLite2-City_[timestamp]" owned by the corresponding user. The second run of the task, started by another user, cannot extract the file to the already existing directory because of the directory ownership.

#### Expected behavior

I would expect multiple runs of the task visit:locate to be independent of each other, not conflicting in the described way.

#### How to reproduce

Just run the task ./cli visit:locate for two different installations with two different users.
Make sure there is no GeoLite2-City.mmdb file in the corresponding data-directory.

The first run succeeds, the second run fails with:
```Downloading GeoLite2 database...
 31546360/31546360 [============================] 100%PHP Warning:  PharData::extractTo(/tmp/GeoLite2-City_20201110/GeoLite2-City.mmdb): failed to open stream: Permission denied in /home/user2/htdocs/vendor/shlinkio/shlink-ip-geolocation/src/GeoLite2/DbUpdater.php on line 67


                                                                                                                        
 [ERROR] GeoLite2 database download failed. It is not possible to locate visits.                                        
                                                                                                                        

                                                                                                                        
 [ERROR] An error occurred while updating geolocation database, and an older version could not be found```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com), and this 
 ### Fixed
 * [#904](https://github.com/shlinkio/shlink/issues/904) Explicitly added missing "Domains" and "Integrations" tags to swagger docs.
 * [#901](https://github.com/shlinkio/shlink/issues/901) Ensured domains which are not in use on any short URL are not returned on the list of domains.
+* [#899](https://github.com/shlinkio/shlink/issues/899) Avoided filesystem errors produced while downloading geolite DB files on several shlink instances that share the same filesystem.
 
 
 ## [2.4.1] - 2020-11-10

--- a/config/autoload/geolite2.global.php
+++ b/config/autoload/geolite2.global.php
@@ -6,8 +6,8 @@ return [
 
     'geolite2' => [
         'db_location' => __DIR__ . '/../../data/GeoLite2-City.mmdb',
-        'temp_dir' => sys_get_temp_dir(),
-        'license_key' => 'G4Lm0C60yJsnkdPi',
+        'temp_dir' => __DIR__ . '/../../data',
+        'license_key' => 'G4Lm0C60yJsnkdPi', // Deprecated. Remove hardcoded license on v3
     ],
 
 ];

